lib/src/presentation/view/movie_list_home.dart

```diff
@@ -1,53 +1,99 @@
 import 'package:flutter/material.dart';
-import '../../data/repository/movie_local_repository.dart';
-import '../../domain/entity/movie.dart';
-import '../../domain/repository/repository_i.dart';
-import '../widget/home_screen/movie_list.dart';
 
 
+import '../../core/util/movie_list_enum.dart';
+import '../../domain/entity/movie_entity.dart';
+import '../bloc/bloc_movie_list.dart';
+import '../widget/home_screen/movie_list.dart';
+import '../widget/home_screen/movie_list_appbar.dart';
+
 class MovieListHome extends StatefulWidget {
-  const MovieListHome({super.key});
+  const MovieListHome({
+    super.key,
+    required this.blocMovieList,
+  });
 
   static const String routeName = "/";
+  final BlocMovieList blocMovieList;
 
   @override
   State<StatefulWidget> createState() => _MovieListHomeState();
 }
 
 class _MovieListHomeState extends State<MovieListHome> {
-  final IRepository<List<Movie>> movieRepository = MovieLocalRepository();
-  late Future<List<Movie>> movies;
+  MovieEndpoint _movieEndpoint = MovieEndpoint.POPULAR;
+
+  void _onDrawerItemTap(MovieEndpoint movieEndpoint) {
+    setState(() {
+      _movieEndpoint = movieEndpoint;
+      widget.blocMovieList.fetchMovies(_movieEndpoint);
+      Navigator.pop(context);
+    });
+  }
 
   @override
   initState() {
     super.initState();
-    movies = movieRepository.loadData();
+    widget.blocMovieList.fetchMovies(_movieEndpoint);
+  }
+
+  @override
+  void dispose() {
+    widget.blocMovieList.dispose();
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      body: FutureBuilder<List<Movie>>(
-        future: movies,
+      appBar: MovieListAppBar(
+        pageTitle: _movieEndpoint,
+      ),
+      drawer: Drawer(
+        child: ListView(
+          children: [
+            DrawerHeader(child: Text('Movies')),
+            ListTile(
+              title: const Text('Popular'),
+              key: Key('PopularDrawer'),
+              onTap: () => _onDrawerItemTap(MovieEndpoint.POPULAR),
```

Comment:
You can wrapped title, key and endpoint on the same enum class

Reply:
You can have something like this:
```
enum MovieEndpoint {
      popular('Popular', 'PopularDrawer');
      
      const MovieEndpoint(this.title, this.key);
      
      final String title;
      final String key;
}
```

Reply:
fixed